src/Lean/Elab/Deriving/BEq.lean

```diff
@@ -57,6 +57,9 @@ where
             let b := mkIdent (← mkFreshUserName `b)
             ctorArgs1 := ctorArgs1.push a
             ctorArgs2 := ctorArgs2.push b
+            let isProof := (← inferType (← inferType x)).isProp
+            if  isProof then
+              continue
             if (← inferType x).isAppOf indVal.name then
               rhs ← `($rhs && $(mkIdent auxFunName):ident $a:ident $b:ident)
             else
```

Comment:
This is running `inferType` twice; maybe store as `xType` and reuse?

Also, maybe use `xType.isProof`:
https://github.com/leanprover/lean4/blob/b614ff1d12bc38f39077f9ce9f2d48b42c003ad0/src/Lean/Meta/InferType.lean#L311-L315

Reply:
Thanks for pointing it out. Looking that the code for `isProof`, I believe it should be `x.isProof` and not `xType.isProof` Since similar code was present in `DecEq.lean`, I've taken the liberty of using `isProof` there too.

Reply:
Ah, I guess I meant `isProof x`? Not sure if it’s better if you have to run `inferType x` later anyways.

Reply:
An good point about `Meta.isProp` and `isProp`. This is very confusing indeed, but I think you got it right :-)

Reply:
My thought process is that `isProofQuick` and `isPropQuick` are very similar, and should be comparable in terms of performance. However, if `isProofQuick` fails, it will have to infer the type, and call `isPropQuick` afterwards. since `inferType x` is already called once, I believe using `isProp` to be more efficient here.